### PR TITLE
Use custom controls for everything

### PR DESCRIPTION
**Pull request by bourdakos1**
## BooleanControl
- coverage (high)
- [x] validation 
    - [x] required check (cannot be required)
    - [x] other validations (none)
- [x] styled

## EnumControl
- coverage (high)
- [x] validation 
    - [x] required check (cannot be required)
    - [x] other validations (none)
- [x] styled

## NumberControl
- coverage (low)
- [x] validation 
    - [x] required check
    - [x] other validations
        - [x] type (`integer` | `number`)
        - [x] multipleOf
        - [x] minimum
        - [x] maximum
        - [x] exclusiveMinimum
        - [x] exclusiveMaximum
- [x] styled

- coverage (high)
- [ ] validation 
    - [x] required check (cannot be required)
    - [ ] other validations
        - [ ] pattern
        - [ ] minLength
        - [ ] maxLength
        - [ ] format
        - [x] uniqueItems
        - [x] minItems
        - [x] maxItems
- [x] styled

## StringControl
- coverage (high)
- [x] validation 
    - [x] required check
    - [x] other validations
        - [x] pattern
        - [x] minLength
        - [x] maxLength
        - [x] format
- [x] styled

### CODE REVIEW

File: packages/pipeline-editor/src/CustomFormControls/EnumControl.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    items,
+    selectedItem: value ?? items[0],
+    onSelectedItemChange: handleSelectedItemChange,
+  });
+
+  return (
+    <Container isOpen={isOpen}>
+      <Button {...getToggleButtonProps()}>
+        <Label>{selectedItem}</Label>
+        <Icon className="elyricon elyricon-chevron-down">
```
Comment (marthacryan) [inline packages/pipeline-editor/src/CustomFormControls/EnumControl.tsx:135]: Should this be `elyraicon`? This is in a few places

Comment (bourdakos1) [reply packages/pipeline-editor/src/CustomFormControls/EnumControl.tsx:135]: it's "elyricon" intentionally, it follows the pattern of other icon packs where the last vowel is dropped in favor of the `i` from icons. vscode has "codicons", github has "octicons"

**Comment by marthacryan** [comment]
When `NumberControl` has `exclusiveMinimum` set to 0 and the field is optional, a validation error appears when there is no value that says "Value must be strictly greater than 0."

**Comment by bourdakos1** [comment]
Oh yea nice catch, I think all the validators will have this issue. I need to disable the check if the field is optional and unset

File: packages/pipeline-editor/package.json [hunk 2]
```diff
@@ -16,11 +16,14 @@
   ],
   "scripts": {
     "build": "microbundle --css inline -f cjs",
-    "watch": "microbundle --css inline -f cjs --watch",
+    "watch": "microbundle --css inline -o dist/index.js --no-pkg-main -f modern --watch --no-compress",
     "link": "yarn link"
   },
+  "DO NOT MERGE THIS IF I FORGOT TO CHANGE THIS BACK, WE ADD FIX THIS FOR REAL IN 'Faster build' PR!!!!": "",
```
Comment (marthacryan) [inline packages/pipeline-editor/package.json:22]: Maybe it's time to remove this?

**Review by marthacryan — APPROVED** [review]
Code LGTM and tested in JupyterLab

**Review by lresende — APPROVED** [review]

**Pull request merged**